doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,471 @@
+
+```sh
+./sbtx assembly
+```
+
+The assembled jar file should be located at
+`target/scala-2.12/spark-exasol-connector-assembly-<VERSION>.jar`.
+
+Then you can use this jar file with `spark-submit`, `spark-shell` or `pyspark`
+commands.
+
+```shell
+spark-shell --jars /path/to/spark-exasol-connector-assembly-*.jar
+```
+
+## Configuration Options
+
+In this section, we describe configuration parameters that are used to
+facilitate the integration between Spark and Exasol clusters.
+
+List of required and optional parameters:
+
+| Spark Configuration           | Configuration    | Default       | Description
+| :---                          | :---             | :---          | :---
+|                               | ``query``        | *<none>*      | An Exasol SQL query string to send to Exasol cluster
+|                               | ``table``        | *<none>*      | A table name (with schema, e.g. schema.table) to save dataframe into
+| ``spark.exasol.host``         | ``host``         | ``10.0.0.11`` | A host ip address to the **first** Exasol node
+| ``spark.exasol.port``         | ``port``         | ``8563``      | A JDBC port number to connect to Exasol database
+| ``spark.exasol.username``     | ``username``     | ``sys``       | An username for connecting to the Exasol database
+| ``spark.exasol.password``     | ``password``     | ``exasol``    | An password for connecting to the Exasol database
+| ``spark.exasol.max_nodes``    | ``max_nodes``    | ``200``       | The number of data nodes in the Exasol cluster
+| ``spark.exasol.batch_size``   | ``batch_size``   | ``1000``      | The number of records batched before running execute statement when saving dataframe
+| ``spark.exasol.create_table`` | ``create_table`` | ``false``     | A permission to create a table if it does not exist in Exasol database when saving dataframe
+| ``spark.exasol.drop_table``   | ``drop_table``   | ``false``     | A permission to drop the table if it exists in Exasol database when saving dataframe
+| ``spark.exasol.jdbc_options`` | ``jdbc_options`` | ``""``        | A string to specify the list of Exasol JDBC options using ``key1=value1;key2=value2`` format
+
+### Max Nodes
+
+Setting the `max_nodes` value to a large number does not increase the connector
+parallelism. The number of parallel connections will always be limited to the
+number of Exasol data nodes.
+
+However, you can use this configuration to decrease the parallelism. This can be
+helpful when debugging an issue. For example, you can set it to one and check if
+the behavior changes.
+
+### JDBC Options
+
+The Spark Exasol Connector uses Exasol JDBC Driver to connect to the Exasol
+cluster from Spark cluster. You can use this configuration parameter to enrich
+the JDBC connection.
+
+For example, to enable debugging with log directory:
+
+```
+.option("jdbc_options", "debug=1;logdir=/tmp/")
+```
+
+Please make sure that it does not start or end with a semicolon (`;`).
```

Comment:
Is it about the directory name?

Reply:
No, it is about the whole options.
Does this better?
```suggestion
Please make sure that the `jdbc_options` string value does not start or end with a semicolon (`;`).
```